builtin/core.wren

```diff
@@ -100,6 +88,29 @@ class Sequence {
   }
 }
 
+class MapSequence is Sequence {
+  new(seq, f) {
+    _seq = seq
+    _f = f
+  }
+
+  iterate(n) { _seq.iterate(n) }
+  iteratorValue(iterator) { _f.call(_seq.iteratorValue(iterator)) }
+}
+
+class WhereSequence is Sequence {
+  new(seq, f) {
+    _seq = seq
+    _f = f
+  }
+
+  iterate(n) {
+    while ((n = _seq.iterate(n)) && !_f.call(_seq.iteratorValue(n))) {}
+    return n
+  }
+  iteratorValue(iterator) { _seq.iteratorValue(iterator) }
```

Comment:
The reason there isn't is because you had not put a blank line either in `MapKeySequence` and `MapValueSequence`, but I will add one. I guess the rule is that there should be a blank line unless the methods are tightly related AND are on one line.